packages/next-app/src/app/metadata/[network]/pool/[poolId]/(components)/MetadataAttributesTable/PredefinedMetadataModal.tsx

```diff
@@ -0,0 +1,196 @@
+        Object.keys(updatedCheckboxes[groupName]).forEach((key) => {
+          updatedCheckboxes[groupName][key] =
+            updatedCheckboxes[groupName]["selectAll"];
+        });
+      } else if (!updatedCheckboxes[groupName][optionName]) {
+        updatedCheckboxes[groupName].selectAll = false;
+      }
+      return updatedCheckboxes;
+    });
+  };
+
+  const getSelectedAttributesAndTypes = (): PoolMetadataAttribute[] => {
+    const selectedAttributes: PoolMetadataAttribute[] = [];
+    for (const attribute of ATTRIBUTES_TEMPLATE) {
+      for (const option of attribute.options) {
+        const { name, type } = option;
+        if (checkboxes[attribute.name][name]) {
+          selectedAttributes.push({
+            typename: type,
+            key: toSlug(name),
+            value: null,
+            description: null,
+          });
+        }
+      }
+    }
+    return selectedAttributes;
+  };
+
+  const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const selectedAttributes = getSelectedAttributesAndTypes();
+    selectedAttributes.map((item) => {
+      if (isKeyUnique(item.key)) {
+        handleAddMetadata(item);
+      }
+    });
+    close?.();
+  };
+
+  return (
+    <form onSubmit={onSubmit}>
+      <Accordion.Root
+        className="my-4 w-full"
+        type="single"
+        defaultValue="item-1"
+        collapsible
+      >
+        {ATTRIBUTES_TEMPLATE.map((attribute) => (
+          <AccordionItem value={attribute.name}>
+            <AccordionTrigger>{attribute.name}</AccordionTrigger>
+            <AccordionContent>
+              <div className="my-0 flex items-center">
+                <Checkbox
+                  id={`select-all-${attribute.name}`}
+                  checked={checkboxes[attribute.name].selectAll}
+                  onChange={() =>
+                    handleCheckboxChange(attribute.name, "selectAll")
+                  }
+                  label={
+                    checkboxes[attribute.name].selectAll
+                      ? "Uncheck all"
+                      : "Check all"
+                  }
+                />
+              </div>
+              <div className="ml-2 flex flex-col">
+                {attribute.options.map((option) => (
+                  <div key={option.name} className="flex items-center">
+                    <Checkbox
+                      id={`${attribute.name}-${option.name}`}
+                      checked={checkboxes[attribute.name][option.name]}
+                      onChange={() =>
+                        handleCheckboxChange(attribute.name, option.name)
+                      }
+                      label={option.name}
+                    />
+                  </div>
+                ))}
+              </div>
+            </AccordionContent>
+          </AccordionItem>
+        ))}
+      </Accordion.Root>
+
+      <div className="flex w-full items-center justify-end gap-2">
```

Comment:
`#thought` we could probably have `Dialog.Footer` to this to 🤔

Reply:
and how we will trigger the submit button to the form?

Reply:
same as now, `Dialog.Footer` has children and we'd have the same buttons here